sway/desktop/render.c

```diff
@@ -430,14 +481,37 @@ static void render_surface_iterator(struct sway_output *output,
 		wlr_output);
 }
 
+static void render_layer_iterator(struct sway_output *output,
+		struct sway_view *view, struct wlr_surface *surface,
+		struct wlr_box *_box, void *_data) {
+	// render the layer's surface
+	render_surface_iterator(output, view, surface, _box, _data);
+
+	struct render_data *data = _data;
+	struct wlr_output *wlr_output = output->wlr_output;
+	struct wlr_box dst_box = *_box;
+	pixman_region32_t *output_damage = data->damage;
+
+	struct decoration_data deco_data = data->deco_data;
+	// TODO: can layers have subsurfaces?
+	bool is_subsurface = view ? view->surface != surface : false;
```

Comment:
pretty sure that null will be passed to this function every time, perhaps we should remove view from the function params, and remove this logic